metadata/rest/orm.py

```diff
@@ -16,14 +16,28 @@ class CherryPyAPI(PacificaModel, ElasticAPI):
     def _select(self, **kwargs):
         """Internal select method."""
         primary_keys = []
+        recursion_depth = CherryPyAPI._check_recursion_depth(kwargs)
         for key in self.get_primary_keys():
             primary_keys.append(getattr(self.__class__, key))
         objs = (self.select()
                 .where(self.where_clause(kwargs))
                 .order_by(*primary_keys))
         if 'page_number' in kwargs and 'items_per_page' in kwargs:
             objs = objs.paginate(int(kwargs['page_number']), int(kwargs['items_per_page']))
-        return dumps([obj.to_hash() for obj in objs])
+        return dumps([obj.to_hash(recursion_depth) for obj in objs])
+
+    @staticmethod
+    def _check_recursion_depth(kwargs):
```

Comment:
I'd like to see something like a ValueError thrown if the recursion depth isn't in [0, 1, 2]
```
recursion_depth = int(kwargs.get('recursion_depth', 1))
if recursion_depth not in range(0, 2):
    raise ValueError("Don't do this you'll hate yourself in the morning.")
```